server/src/main/java/org/elasticsearch/action/admin/cluster/configuration/TransportClearVotingConfigExclusionsAction.java

```diff
@@ -105,7 +106,13 @@ public void onTimeout(TimeValue timeout) {
 
     private void submitClearVotingConfigExclusionsTask(ClearVotingConfigExclusionsRequest request, long startTimeMillis,
                                                        ActionListener<ClearVotingConfigExclusionsResponse> listener) {
-        clusterService.submitStateUpdateTask("clear-voting-config-exclusions", new ClusterStateUpdateTask(Priority.URGENT) {
+        final long timeout = request.getTimeout().millis() + startTimeMillis - threadPool.relativeTimeInMillis();
```

Comment:
I wonder if we should just do Math.max(0, timeout) instead, just like in cluster health action? Seems like the right place to handle timeout=0 is inside `submitStateUpdateTask`, rather than special handle it here?

Reply:
++ I pushed 91e5fb3